crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -2189,6 +2189,20 @@ where
                     }
                 }
 
+                // Ex) Union[...]
+                if self.enabled(Rule::UnnecessaryLiteralUnion) {
+                    // Avoid duplicate checks if the parent is an `Union[...]`
+                    if self.semantic.expr_parent().map_or(true, |parent| {
+                        if let Expr::Subscript(ast::ExprSubscript { value, .. }) = parent {
+                            !self.semantic.match_typing_expr(value, "Union")
+                        } else {
+                            false
+                        }
+                    }) {
```

Comment:
Nit: I think this may work
```suggestion
                    if let Some(Expr::Subscript(ast::ExprSubscript { value, .. })) = self.semantic.expr_parent()
                         !self.semantic.match_typing_expr(value, "Union")
                    } else {
                   	false
                    } {
```